### PR TITLE
Extended SetAttribute() method to include all integer types

### PR DESCRIPTION
**Pull request by snehilchopra**
Added code to cover all integer types for the GCP recordable.
Added test coverage for the same.

### CODE REVIEW

File: exporters/trace/gcp_exporter/recordable.h [hunk 1]
```diff
@@ -22,16 +22,26 @@ class Recordable final : public sdk::trace::Recordable
...
+                      opentelemetry::trace::SpanId parent_span_id) noexcept override;
 
   void SetAttribute(nostd::string_view key,
-                    const opentelemetry::common::AttributeValue &&value) noexcept override;
+                            const opentelemetry::common::AttributeValue &value) noexcept override;
 
-  void AddEvent(nostd::string_view name, core::SystemTimestamp timestamp) noexcept override;
+  void AddEvent(
+      nostd::string_view name,
+      core::SystemTimestamp timestamp = core::SystemTimestamp(std::chrono::system_clock::now()),
```
Comment (nadiaciobanu) [inline exporters/trace/gcp_exporter/recordable.h:34]: I don't think these defaults are necessary. Once https://github.com/open-telemetry/opentelemetry-cpp/pull/208 merges, all defaults will be handled using function overloading in the base `Recordable` class. The same applies to `AddLink()`.

File: exporters/trace/gcp_exporter/internal/recordable_test.cc [hunk 2]
```diff
@@ -16,12 +16,30 @@ TEST(Recordable, TestSetAttribute)
...
+    // Set 'int' type
     const nostd::string_view int_key = "int_key";
-    const int64_t seven = 7;
-    const common::AttributeValue int_value = seven;
+    const int seven_int = 7;
+    const common::AttributeValue int_value = seven_int;
     rec.SetAttribute(int_key, std::move(int_value));
 
+    // Set 'int64_t' type
+    const nostd::string_view int64_t_key = "int64_t_key";
```
Comment (nadiaciobanu) [inline exporters/trace/gcp_exporter/internal/recordable_test.cc:26]: You may consider using templates here to reduce code duplication (see [OTLP Recordable test](https://github.com/open-telemetry/opentelemetry-cpp/blob/60f6920acb94089c903aaa9c822c7cbb7125554c/exporters/otlp/test/recordable_test.cc#L197)). It's totally optional though.

File: exporters/trace/gcp_exporter/internal/recordable.cc [hunk 3]
```diff
@@ -43,6 +42,22 @@ void Recordable::SetAttribute(nostd::string_view key,
     {
         (*map)[std::string(key)].set_bool_value(nostd::get<bool>(value));
     } 
+    else if (nostd::holds_alternative<int>(value))
+    {
+        (*map)[std::string(key)].set_int_value(nostd::get<int>(value));
```
Comment (nadiaciobanu) [inline exporters/trace/gcp_exporter/internal/recordable.cc:47]: I think you can also use `key.data()` here, which might be more clear.

File: exporters/trace/gcp_exporter/internal/recordable.cc [hunk 4]
```diff
@@ -43,6 +42,22 @@ void Recordable::SetAttribute(nostd::string_view key,
...
+        (*map)[std::string(key)].set_int_value(nostd::get<int64_t>(value));
+    }
+    else if (nostd::holds_alternative<unsigned int>(value))
+    {
+        (*map)[std::string(key)].set_int_value(nostd::get<unsigned int>(value));
+    }
+    else if (nostd::holds_alternative<uint64_t>(value))
+    {
+        (*map)[std::string(key)].set_int_value(nostd::get<uint64_t>(value));
+    }
```
Comment (nadiaciobanu) [inline exporters/trace/gcp_exporter/internal/recordable.cc:60]: Is there any plan to support array attribute types? Or does GCP not support that?

File: exporters/trace/gcp_exporter/recordable.h (hunk 1 above)
Comment (snehilchopra) [reply exporters/trace/gcp_exporter/recordable.h:34]: Oh I see.
Alright. Thanks!

File: exporters/trace/gcp_exporter/internal/recordable_test.cc (hunk 2 above)
Comment (snehilchopra) [reply exporters/trace/gcp_exporter/internal/recordable_test.cc:26]: Yes, I was considering that. Just trying to contain all the different types( string, bool as well as all int types) in one single test.

@IlyaKobelevskiy @ziweizhao thoughts?

File: exporters/trace/gcp_exporter/internal/recordable.cc (hunk 4 above)
Comment (snehilchopra) [reply exporters/trace/gcp_exporter/internal/recordable.cc:60]: [Official docs](https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.cloudtrace.v2#google.devtools.cloudtrace.v2.Span.Attributes) say it only takes in three types: bool, int, and string.

File: exporters/trace/gcp_exporter/recordable.h (hunk 1 above)
Comment (snehilchopra) [reply exporters/trace/gcp_exporter/recordable.h:34]: I am assuming I'll have to wait for it to be merged and then run my tests?

File: exporters/trace/gcp_exporter/internal/recordable.cc [hunk 5]
```diff
@@ -34,31 +33,60 @@ void Recordable::SetIds(trace::TraceId trace_id,
...
 
 
-void Recordable::AddEvent(nostd::string_view name, core::SystemTimestamp timestamp) noexcept
+void Recordable::AddEvent(nostd::string_view name, 
+                          core::SystemTimestamp timestamp,
+                          const opentelemetry::trace::KeyValueIterable &attributes) noexcept
 {
     (void)name;
+    (void)timestamp;
+    (void)attributes;
```
Comment (snehilchopra) [inline exporters/trace/gcp_exporter/internal/recordable.cc:80]: Should I go with unnamed parameters here and below? 
Not sure what the specification is under the GCP org for unused parameters.

Comment (IlyaKobelevskiy) [reply exporters/trace/gcp_exporter/internal/recordable.cc:80]: I am not sure what is the spec, but I think those parameters are unused for a short term.
Eventually this code AddEvent should eventually map to a TimeEvent (https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.cloudtrace.v2#timeevent).

**Pull request merged**